data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -127,7 +186,7 @@ private void compileMatchPatterns() {
 
     private void mergeCaptures(final Map<String, Object> original, final Map<String, Object> updates) {
         for (final Map.Entry<String, Object> updateEntry : updates.entrySet()) {
-            if (!(original.containsKey(updateEntry.getKey()))) {
+            if (!(original.containsKey(updateEntry.getKey())) || grokPrepperConfig.getOverwrite().contains(updateEntry.getKey())) {
```

Comment:
This is `List`. If we are going to keep the current `overwrite` behavior, we also need to make this a `Set`, at least in this class. Maybe not in the `GrokPrepperConfig`.

Reply:
Just create a new `HashSet<>(grokPrepperConfig.getOverwrites())` in the constructor for `GrokPrepper`.